app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,71 @@
+class QuestionsController < ApplicationController
+
+ skip_before_action :logged_in?, only: [:index, :show]
+ include VotesHelper
+
+  def new
+    @question = Question.new
+  end
+
+  def create
+    @question = Question.new(question_params.merge(user: current_user))
+    if @question.save
+      @question.set_tags(@question.tag_string)
+      redirect_to question_path(@question)
+    else
+      flash[:notice] = "You need to fill out both fields. Please try again"
+    end
+  end
+
+  def edit
+    @question = Question.find(params[:id])
+  end
+
+  def update
+    @question = Question.find(params[:id])
+    redirect_to "/" unless @question.is_owned_by(current_user)
```

Comment:
Also consider using before_action to implement the ownership restrictions - ie redirect before reaching the action.

So you could add a before action that redirects with a flash message unless the current_user is the owner of the question and attach it to update, edit, destroy.